docs/docs/reference/release-notes/image-migration-guide.md

```diff
@@ -0,0 +1,214 @@
+
+Note that if you're trying to do a partial migration because some uses of `gatsby-image` are relying on CMS data that isn't compatible yet, you'll want to run the codemod on individual files rather than your whole project.
+
+There are two ways to run the codemod. Most users will want to use the `npx` version.
+
+```shell
+npx gatsby-codemods gatsby-plugin-image <optional-path>
+
+```
+
+Without a path, the codemod will run against all the files in your current directory, so running it in root is recommended. It will ignore `node_modules`, `.cache` and `public` automatically. It will also respect any local Babel configuration in your project. The codemod is designed to run against files with the extensions `.ts, .js, .tsx, .jsx`. If this does not cover your project, or you require other customizations, you'll want to run the codemod using jscodeshift.
+
+1. Install jscodeshift
+
+```shell
+npm install --global jscodeshift
+```
+
+2. Install the codemods package in your project
+
+```shell
+npm install gatsby-codemods
+```
+
+3. Run the codemod
+
+```shell
+jscodeshift -t node_modules/gatsby-codemods/transforms/gatsby-plugin-image.js .
+```
+
+See the [jscodeshift docs](https://github.com/facebook/jscodeshift) for all the available flags.
+
+### Codemod Assumptions
+
+Due to the API changes, the codemod is not a pure 1:1 mapping. There are some changes introduced.
+
+1. Fluid images will map to either fullWidth or constrained images. This decision is made based on the existence of maxWidth and its value. If maxWidth does not exist, it will be a fullWidth image. If it does, and the maxWidth is less than 1000, it will be a constrained image, otherwise a fullWidth image. fullWidth images do not retain their maxWidth or maxHeight fields, constrained images do, as width and height.
+2. All images will generate Webp
+
+The API change to options is not currently codemoded. TODO: should this be?
+
+The codemod will warn the user in a number of different scenarios and point you to the file in question so you can inspect the changes manually.
+
+### Recommended Changes
+
+For images using static query, you should move to use the `StaticImage` component instead. This component takes `src` which can be a remote image URL or a relative path to an image. Make sure you've installed `gatsby-source-filesystem` if you're going to use this component.
+
+The plugin also includes helper functions as the structure of the gatsbyImageData object is considered an implementation detail.
+
+```javascript
+import { getImage, getSrc, GatsbyImage } from "gatsby-plugin-image"
+
+const HomePage = ({ data }) => {
+  const image = getImage(data.file)
+  const imagePath = getSrc(data.file)
+  return (
+    <div>
+      <GatsbyImage image={image} alt="please include an alt" />
+      <SEO imageSrc={imagePath} />
+    </div>
+  )
+}
+```
+
+### Unsupported gatsby-image usage
```

Comment:
I wonder if this section makes more sense up with the New Syntax / API changes sections?

Also are these considered "breaking changes"? I wonder if that wording is more of an attention grabber?

Reply:
Breaking depending on the user, but for sure.